fastcrypto/src/groups/secp256k1.rs

```diff
@@ -199,25 +273,446 @@ impl ScalarTrait for Scalar {
 
 impl ToFromByteArray<SCALAR_SIZE_IN_BYTES> for Scalar {
     fn from_byte_array(bytes: &[u8; SCALAR_SIZE_IN_BYTES]) -> Result<Self, FastCryptoError> {
-        Ok(Scalar(
-            Fr::deserialize_uncompressed(bytes.as_slice())
-                .map_err(|_| FastCryptoError::InvalidInput)?,
-        ))
+        // Align serialization with the k256 crate which follows the Bitcoin implementation
+        // TODO: Check canonical repr
+        Ok(Scalar::from_bytes_mod_order(bytes))
     }
 
     fn to_byte_array(&self) -> [u8; SCALAR_SIZE_IN_BYTES] {
-        let mut bytes = [0u8; SCALAR_SIZE_IN_BYTES];
-        self.0
-            .serialize_uncompressed(&mut bytes[..])
-            .expect("Byte array not large enough");
-        bytes
+        self.0.into_bigint().to_bytes_be().try_into().unwrap()
     }
 }
 
-impl ToLittleEndianBytes for Scalar {
-    fn to_le_bytes(&self) -> Vec<u8> {
-        self.to_byte_array().to_vec()
+serialize_deserialize_with_to_from_byte_array!(Scalar);
+
+/// A Schnorr signature scheme as defined in BIP-0340 (https://github.com/bitcoin/bips/blob/master/bip-0340.mediawiki).
+pub mod schnorr {
```

Comment:
I'm not sure how much of this we are going to use, but we can mark it test-only until we know.